arclightning_backend/src/splash.rs

```diff
@@ -0,0 +1,16 @@
+extern crate rand;
+use rand::Rng;
+use std::fs;
+use std::io;
+
+fn select_random_image(image_dir: PathBuf) -> io::Result<PathBuf>{
+
+    
+    let mut paths: Vec<PathBuf> = fs::read_dir("image_dir").?;
+    let path = rand::thread_rng().choose(&paths)?;
```

Comment:
`?` doesn't technically work here, because `choose` returns an `Option`
You should add `.ok_or_else(|| io::Error::new(io::ErrorKind::Other, "choosing random file failed"))` to the end, and remove the `?`. This is because `?` is good for exiting due to error early, however we want to return a `Result` anyway, so we can skip the `?` (which returns if there is an error)